### DISCUSSION TITLE
How do I use mini.deps instead of cloning the entire mini.nvim repository.

### DISCUSSION THREAD

**Discussion by 231tr0n** [nvim-mini/mini.nvim#1175] [Q&A] [answered]
```lua
-- MiniDeps auto download and configure setup
local path_package = vim.fn.stdpath("data") .. "/site/"
local mini_path = path_package .. "pack/deps/start/mini.deps"
if not vim.uv.fs_stat(mini_path) then
	vim.cmd('echo "Installing `mini.deps`" | redraw')
	local clone_cmd = {
		"git",
		"clone",
		"--filter=blob:none",
		"https://github.com/echasnovski/mini.deps",
		mini_path,
	}
	vim.fn.system(clone_cmd)
	vim.cmd('echo "Installed `mini.deps`" | redraw')
	vim.cmd("packadd mini.deps | helptags ALL")
end
require("mini.deps").setup({ path = { package = path_package } })

-- add, now and later functions from MiniDeps
local add, now, later = MiniDeps.add, MiniDeps.now, MiniDeps.later
```

Is this the right way to use mini.deps without cloning entire mini.nvim or am I getting something wrong?
Also should I be calling ``add("echasnovski/mini.deps")`` after the above setup?

**Comment by echasnovski** [comment]
Replacing 'mini.nvim' with 'mini.deps' in [the initial portion of the example 'init.lua'](https://github.com/echasnovski/mini.nvim/blob/2c55015a518e11ea9d49bb96078f5ede51f5d2af/scripts/init-deps-example.lua#L1-L13) should be enough. Which seems what is done here. Do you have actual problems with this setup?

**Reply by 231tr0n** [reply]
Every now and then I get this error. It is inconsistent though.

![image](https://github.com/user-attachments/assets/2be162dd-48e2-497e-aacd-0cffef850534)

**Reply by 231tr0n** [reply]
Do I have to ``add("echasnovski/mini.deps")`` after mini.deps setup function is called?

**Reply by echasnovski** [reply, marked as answer]
> Every now and then I get this error. It is inconsistent though.

I am not sure what this is about. I tried `:DepsUpdate` several times after several restarts with the following 'init.lua' and everything worked as expected:
```lua
-- MiniDeps auto download and configure setup
local path_package = vim.fn.stdpath("data") .. "/site/"
local mini_path = path_package .. "pack/deps/start/mini.deps"
if not vim.uv.fs_stat(mini_path) then
	vim.cmd('echo "Installing `mini.deps`" | redraw')
	local clone_cmd = {
		"git",
		"clone",
		"--filter=blob:none",
		"https://github.com/echasnovski/mini.deps",
		mini_path,
	}
	vim.fn.system(clone_cmd)
	vim.cmd('echo "Installed `mini.deps`" | redraw')
	vim.cmd("packadd mini.deps | helptags ALL")
end
require("mini.deps").setup({ path = { package = path_package } })

-- add, now and later functions from MiniDeps
local add, now, later = MiniDeps.add, MiniDeps.now, MiniDeps.later

now(function()
	add("echasnovski/mini.notify")
	require("mini.notify").setup()
end)

now(function()
	add("echasnovski/mini.hues")
	require("mini.hues").setup({ background = "#111111", foreground = "#dddddd" })
end)
```

To me it looks like 'mini.deps' was somehow altered from the outside. Maybe try removing everything and reinstalling?

> Do I have to `add("echasnovski/mini.deps")` after mini.deps setup function is called?

The [example 'init.lua'](https://github.com/echasnovski/mini.nvim/blob/2c55015a518e11ea9d49bb96078f5ede51f5d2af/scripts/init-deps-example.lua#L1-L13) doesn't have `add('echasnovski/mini.nvim')` line, so there should be no need for `add('echasnovski/mini.deps)`.

**Reply by 231tr0n** [reply]
@echasnovski seems like what you told does the job. Thank you soo much.
